clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1027,9 +1027,10 @@ public int numStagedReceives(KafkaChannel channel) {
 
     private class SelectorMetrics implements AutoCloseable {
         private final Metrics metrics;
-        private final String metricGrpPrefix;
```

Comment:
i understand this no longer geets used but in the interesting of minimizing our diff to upstream i'd keep it. up to you though

Reply:
Upstream seems to be reviewing PR already. I will wait for a day or two to see if it gets merged and adjust this patch accordingly.